apps/ledger-live-mobile/src/components/Stake/index.tsx

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable consistent-return */
+import { useMemo, useLayoutEffect, useCallback } from "react";
+import { StackNavigationProp } from "@react-navigation/stack";
+import { useNavigation } from "@react-navigation/native";
+import { Account } from "@ledgerhq/types-live";
+import {
+  listCurrencies,
+  filterCurrencies,
+} from "@ledgerhq/live-common/currencies/helpers";
+import { useFeature } from "@ledgerhq/live-common/featureFlags/index";
+import { NavigatorName, ScreenName } from "../../const";
+import perFamilyAccountActions from "../../generated/accountActions";
+import logger from "../../logger";
+
+const StakeFlow = () => {
+  const featureFlag = useFeature("stakePrograms");
+  const list = featureFlag?.params?.list;
+  const navigation = useNavigation();
+  const cryptoCurrencies = useMemo(() => {
+    return filterCurrencies(listCurrencies(true), {
+      currencies: list || [],
+    });
+  }, [list]);
+
+  const onSuccess = useCallback(
+    (account: Account, parentAccount?: Account) => {
+      // @ts-expect-error issue in typing
+      const decorators = perFamilyAccountActions[account?.currency?.family];
+      const familySpecificMainActions =
+        (decorators &&
+          decorators.getMainActions &&
+          decorators.getMainActions({
+            account,
+            parentAccount,
+            colors: {},
+          })) ||
+        [];
+      const stakeFlow = familySpecificMainActions.find(
+        (action: { id: string }) => action.id === "stake",
+      )?.navigationParams;
+      if (!stakeFlow) return null;
+      const [name, options] = stakeFlow;
+
+      // TODO: Remove after Kiln stake implementation was be done
+      if (
+        account?.currency?.family === "ethereum" &&
+        name === NavigatorName.Base
+      ) {
+        navigation.navigate(name, {
+          screen: options.screen,
+          params: {
+            ...(options?.params || {}),
+            accountId: account?.id,
+            parentId: parentAccount?.id,
+          },
+        });
+      } else {
+        (
+          navigation as StackNavigationProp<{
```

Comment:
I think it's best to avoid casting `navigation`, it could hide some possible issues. I assume typescript is complaining about the params passed to `navigation.navigate`?

You might be able to do this instead:

```
const navigation = useNavigation<StackNavigationProp<{ [key: string]: object | undefined }>>();
```